js/gallery/shared/Gallery.svelte

```diff
@@ -54,16 +56,30 @@
 
 	$: was_reset = value == null || value.length === 0 ? true : was_reset;
 
-	let resolved_value: GalleryData | null = null;
-	$: resolved_value =
-		value == null
-			? null
-			: value.map((data) => ({
-					image: data.image as FileData,
-					caption: data.caption
-				}));
-
-	let prev_value: GalleryData | null = value;
+	let resolved_value: GalleryData[] | null = null;
+	let thumbnailsGenerated = false;
```

Comment:
@dawoodkhan82 this implementation delivers but why do we need to manually generate thumbnails?

I think we can display videos inside the buttons like so

```svelte
<Video
  src={media.video.url}
  title={media.caption || null}
  is_stream={false}
  data-testid={"thumbnail " + (i + 1)}
  alt=""
  loading="lazy"
  loop={false}
/>
```

Reply:
Yeah originally had it this way, but thought loading videos would be too intensive. Looking at it again, i think you're right. Let me revert